### ISSUE TITLE
[Question] USB Keyboard for LUKS on steam deck

### ISSUE THREAD

**Issue by FearlessSpiff** [pmkap/deckrypt#1]
Hi. Did you ever get this running on the deck itself? It looks really promising, but I am stuck as I cant even use my keyboard to enter the password for LUKS decryption. Does this work for you? Thanks!

**Comment by FearlessSpiff** [comment]
Fix was to change USB mode from DRD to XHCI in Bios. So now I will be able to check your code. :)

**Comment by pmkap** [comment]
Hi @FearlessSpiff 

To be honest, I never tried it with the Steam Deck because so far I didn't get one. But I ordered a Steam Deck OLED so I will try soon!

> So now I will be able to check your code. :)

Let me know how it goes!

**Comment by TobiPeterG** [comment]
@pmkap 
Hey, I'll try this as well, is it compatible with other handheld devices as well or would it need additional work to support different button layouts etc?
Or would it support all standard gamepads?

**Comment by pmkap** [comment]
> @pmkap Hey, I'll try this as well, is it compatible with other handheld devices as well or would it need additional work to support different button layouts etc? Or would it support all standard gamepads?

Yes, this should work with every gamepad as long as the gamepad's buttons are defined in this list https://github.com/pmkap/deckrypt/blob/edbf8c0ae5e0a09b722a80ec7f5f3e648d2305cc/src/deckrypt_input.c#L49

You can run `evtest` to find them.

**Comment by TobiPeterG** [comment]
Hey, you all feel free to check out my fork:
https://github.com/TobiPeterG/deckrypt

I ported the code to Rust and made several improvements (less crashes, controller profiles, automatic button mapping, controller detection, controller profiles, ...)
I can already use it to unlock my root disk with my controller on openSUSE ;)
